### ISSUE TITLE
Theories are run exponentially times per InlineData entry

### ISSUE THREAD

**Issue by kemmis** [MinistryOfMagic/xunit.frameworks.autofac#3]
Did nobody notice that theories are running multiple times per InlineData? 

Reproduce:

```
 [UseAutofacTestFramework]
    [Collection("Foo")]
    public class CollectionFixtureTests
    {
        private static int _numRuns = 0;

        [Theory]
        [InlineData("ABC-02-04-CDE")]
        [InlineData("ABC-02-40-CDE")]
        [InlineData("ABC-02-0040-CDE")]
        public void Theories_work_correctly(string input)
        {
            input.Should().Be(input);

            _numRuns++;
            Assert.True(_numRuns <= 3, $"WHY U RUN MUCH TIMES? U RUN {_numRuns} ALREADY.");
        }
    }
```

Expected behavior: _numRuns should always be <=3 since there are 3 InlineData entries
Actual behavior: test runs 6 times and assert fails:

![image](https://user-images.githubusercontent.com/647545/120071182-0ea2b300-c05c-11eb-89d9-486377242664.png)

It's not always apparent in the regular VS TestExplorer, but with the Resharper test explorer you see that there's some wild logic going on here:

![image](https://user-images.githubusercontent.com/647545/120071291-9b4d7100-c05c-11eb-8d32-df03afc22a4e.png)

**Comment by kemmis** [comment]
NEW PR: https://github.com/MinistryOfMagic/xunit.frameworks.autofac/pull/8

**Comment by dabide** [comment]
Fixed in #8
